### ISSUE TITLE
Cannot build wasabi

### ISSUE THREAD

**Issue by jlab55** [intuit/wasabi#283]
Hello,
I am having a mess of a time just building this project. After cloning the repo and then bootstrapping (everything ran fine there) the actual build itself does not work. (I am on Mac OS Sierra 10.12.6)

There were numerous API compatibility issues maven complains about. I resolved them in the root pom.xml via:
```
<artifactId>maven-jar-plugin</artifactId>
-                <version>2.6</version>
+                <version>3.0.2</version>
<artifactId>maven-javadoc-plugin</artifactId>
-                <version>2.10.3</version>
+                <!--<version>2.10.3</version>-->
                 <artifactId>maven-source-plugin</artifactId>
-                <version>2.4</version>
+                <version>3.0.1</version>
```

That gets me a little further in the build process (where modules are actually JAR'ing). However, wasabi-repository-datastax fails to build:

```
[ERROR] Failed to execute goal on project wasabi-repository-datastax: Could not resolve dependencies for project com.intuit.wasabi:wasabi-repository-datastax:jar:1.0.20170829225429-SNAPSHOT: The following artifacts could not be resolved: com.intuit.wasabi:wasabi-cassandra-datastax:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-experiment-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-authorization-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-authentication-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-analytics-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-user-directory:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-feedback-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-auditlog-objects:jar:1.0.20170829225429-SNAPSHOT, com.intuit.wasabi:wasabi-database:jar:1.0.20170829225429-SNAPSHOT: Could not find artifact com.intuit.wasabi:wasabi-cassandra-datastax:jar:1.0.20170829225429-SNAPSHOT in hack.repository (file:/Users/JL/exp/wasabi/modules/repository-datastax/../../repository) -> [Help 1]
```
I am confused why it is not picking up the previous JARs that were built (e.g. for dependency wasabi-cassandra-datastax:jar:1.0.20170829225429-SNAPSHOT the file wasabi-cassandra-datastax-1.0.20170829225429-SNAPSHOT.jar is in modules/cassandra-datastax/target

I even tried doing a maven install in the modules/cassandra-datastax to get into my local repo but that did not help.

Any guidance on how to build the project would be greatly appreciated! My goal is to create an auth class that uses LDAP instead of a simple file. I would be happy to share that back to the community if I could just get the project to build out of the box.

Thanks,
Jon

**Comment by jlab55** [comment]
@abhishekjain88 or @abheekd would you have any ideas as to why I cannot build wasabi? Seems like all the JARs from the previous modules are not showing up on the classpath when wasabi-repository-datastax is building. I really appreciate any help you can offer!!! Thanks, Jon

**Comment by jlab55** [comment]
I have found and fixed the problem. The referenced errors are extremely misleading. The issue is actually that I have two JVMs installed and this project and it's associated maven plugins are highly dependant on 1.8.0_144. 

Bootstrapping correctly downloads them but if the JAVA_HOME was already set it does not override. I wrote a simple bash script to check during the build.sh execution to verify the correct JVM is installed (e.g. checking for java -version output for 1.8.0_144) and will propose it as a merge.
